server/server.py

```diff
@@ -0,0 +1,566 @@
+# Network related functions
+# **********************************************************************
+def recv_packet(conn):
+    """
+    Receive and unpack a packet from the connection.
+    - Reads the packet header to determine payload size, then retrieves and unpacks the full packet.
+    """
+    header = conn.recv(HEADER_SIZE)  # Receive the fixed-size header
+    payload_size = RequestUnpacker.get_payload_size(header)  # Get payload size from header
+    # Combine header and payload, then unpack the packet
+    packet = RequestUnpacker.RequestPacketUnpacker(header + conn.recv(payload_size)).unpack()
+    return packet  # Return the unpacked packet
+
+def send_packet(conn, packet):
+    """
+    Send a packet over the connection.
+    - Sends the fully formed packet data to the connected peer.
+    """
+    conn.sendall(packet)  # Send the entire packet to the destination
+
+# **********************************************************************
+# Server start-up functions
+# **********************************************************************
+def handle_registration(packet, conn):
+    """
+    Handles new client registration.
+    - Generates a UUID for the client, checks for existing registration, and stores the new client in the database.
+    - Sends success or failure response based on registration outcome.
+    """
+    client_name = packet['name']
+    print(f"Got a registration request, client name: {client_name}")
+
+    # Generate a unique identifier for the client
+    client_id = uuid.uuid4().bytes
+    
+    # Check if the client name is already registered in the database
+    with sqlite3.connect("defensive.db") as conn_db:
+        cursor = conn_db.cursor()
+        cursor.execute("SELECT ID FROM clients WHERE Name = ?", (client_name,))
+        result = cursor.fetchone()
+        
+        if result:
+            # Client already registered, send failure response
+            print(f"Client {client_name} is already registered, sending failure packet.")
+            reg_fail_pckt = ResponsePacker.RegistrationFailedPacker(version=SERVER_VER)
+            conn.sendall(reg_fail_pckt.pack())
+            return ERROR
+        else:
+            # Register the new client by inserting into the database
+            cursor.execute('''INSERT INTO clients (ID, Name, LastSeen) VALUES (?, ?, ?)''', 
+                           (client_id, client_name, datetime.now()))
+            conn_db.commit()
+
+    # Add client to in-memory storage
+    clients_dict[client_name] = Client(name=client_name, id=client_id)
+
+    # Send success response with the assigned UUID
+    reg_success_pckt = ResponsePacker.RegistrationSucceededPacker(version=SERVER_VER, client_id=client_id)
+    conn.sendall(reg_success_pckt.pack())
+    print(f"Registration successful for {client_name}. Assigned UUID: {client_id.hex()}")
+
+    # Wait for public key packet
+    packet = recv_packet(conn) 
+    if packet['code'] != 826:

```

Comment:
magic.
imagine how much more readable this couldve been if this read: `packet.code != PUBLIC_KEY_REQUEST`